compiler/dyno/lib/uast/Builder.cpp

```diff
@@ -280,12 +289,118 @@ void Builder::doAssignIDs(AstNode* ast, UniqueString symbolPath, int& i,
   if (search != notedLocations_.end()) {
     assert(!search->second.isEmpty());
     idToLocation_[ast->id()] = search->second;
+    if (ieNode) {
+      checkConfigPreviouslyUsed(ast, configNameUsed);
+    }
   } else {
     assert(false && "Location for all ast should be set by noteLocation");
   }
 }
 
-AstList Builder::flattenTopLevelBlocks(AstList lst) {
+  void Builder::checkConfigPreviouslyUsed(const AstNode* ast, std::string& configNameUsed) {
+    parsing::useConfigParam(context(), configNameUsed, ast->id());
+    auto usedId = parsing::nameToConfigParamId(context(), configNameUsed);
+
+    if (usedId != ast->id()) {
+      std::string err = "ambiguous config name (";
+      err += configNameUsed;
+      err += ")";
+      std::string otherLoc = "also defined here";
+      std::string disambiguate = "(disambiguate using -s<modulename>.";
+      disambiguate += configNameUsed;
+      disambiguate += "...)";
+      ErrorMessage errorMessage = ErrorMessage(ast->id(), idToLocation_[ast->id()], err, ErrorMessage::ERROR );
+      ErrorMessage noteOtherLoc = ErrorMessage(ast->id(), idToLocation_[usedId], otherLoc, ErrorMessage::NOTE);
+      ErrorMessage noteDisambiguate = ErrorMessage(ast->id(), idToLocation_[usedId], disambiguate, ErrorMessage::NOTE);
+      addError(errorMessage);
+      addError(noteOtherLoc);
+      addError(noteDisambiguate);
+    }
+  }
+
+  std::tuple<AstNode*,std::string> Builder::checkAndUpdateConfig(AstNode* ast, pathVecT& pathVec) {
```

Comment:
IMO it would make more sense to use `std::pair` here. But obviously it could work either way. Is there a particular reason for `std::tuple` ?

Also I would probably add a reference argument to return one of these through rather than returning a pair/tuple. I don't object to a pair/tuple though.

Reply:
I changed this around to pass config name and value as ref args instead of returning pairs or tuples